maphooks/useBreadcrumbs.js

```diff
@@ -27,19 +28,56 @@ export default function useBreadcrumbs(aois, viewport) {
             }
         }
     
-        const parents = aois.locations.filter(x => !x.parent).filter(p => testContains(p.location_awareness.bbox, viewport))
+        const parents = aois.filter(x => !x.parent).filter(p => testContains(p.location_awareness.bbox, viewport))
         if (parents.length === 0) {
             return []
         }
     
-        const children = aois.locations.filter(c => c.parent === parents[0].id).filter(c => testContains(c.location_awareness.bbox, viewport))
-        if (children.length === 0) return [parents[0].id]
-        return [parents[0].id, children[0].id]
+        const children = aois.filter(c => c.parent === parents[0].id).filter(c => testContains(c.location_awareness.bbox, viewport))
+        if (children.length === 0) return [parents[0]]
+        return [parents[0], children[0]]
     }
 
     useEffect(() => {
         setBreadcrumbs(getBreadcrumbs(viewport))
     }, [viewport]) // eslint-disable-line react-hooks/exhaustive-deps
 
     return breadcrumbs
+}
+
+export function useMapWithBreadcrumbs(viewport, aois, map) {
+    const [aoisToPlace, setAoisToPlace] = useState([])
+    const [markers, setMarkers] = useState([])
+
+    useEffect(() => {
+        const filtered_aois = aois.filter(aoi => {
+            return viewport.zoom >= aoi.location_awareness.minzoom && viewport.zoom <= aoi.location_awareness.maxzoom
+        })
+        setAoisToPlace(filtered_aois)
+    }, [viewport])
```

Comment:
Should aois be a dependency here?

Reply:
I don't think they need to be, since they're static.  We don't expect new ones at runtime.